File: app/src/main/java/me/magnum/melonds/model/Rom.java
package me.magnum.melonds.model;

public class Rom {
	private String name;
	private String path;
	private RomConfig config;

	public Rom(String name, String path, RomConfig config) {
		this.name = name;
		this.path = path;
		this.config = config;
	}

	public String getName() {
		return this.name;
	}

	public String getPath() {
		return this.path;
	}

	public RomConfig getConfig() {
		return config;
	}

	public void setConfig(RomConfig config) {
		this.config = config;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;

		Rom rom = (Rom) o;

		return path.equals(rom.path);
	}

	@Override
	public int hashCode() {
		return path.hashCode();
	}
}

File: app/src/main/java/me/magnum/melonds/model/Rom.kt
package me.magnum.melonds.model

import java.util.*

data class Rom(val name: String, val path: String, var config: RomConfig, var lastPlayed: Date? = null) {
    override fun equals(other: Any?): Boolean {
        if (this === other)
            return true

        if (other == null || javaClass != other.javaClass)
            return false

        val rom = other as Rom
        return path == rom.path
    }

    override fun hashCode(): Int {
        return path.hashCode()
    }
}